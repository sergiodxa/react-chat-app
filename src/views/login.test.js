import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Login from "./login";

test("Login component", () => {
  const onUsername = jest.fn();

  const { asFragment, getByLabelText } = render(
    <Login onUsername={onUsername} />
  );

  expect(asFragment()).toMatchSnapshot();

  const input = getByLabelText("Enter your username");

  fireEvent.change(input, {
    target: { value: "test-user" }
  });

  fireEvent.submit(input);

  expect(onUsername).toHaveBeenCalledWith("test-user");
});
