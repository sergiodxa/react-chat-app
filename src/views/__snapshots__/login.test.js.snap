// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component 1`] = `
<DocumentFragment>
  .emotion-5 {
  background: #CAD1E0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  width: 100%;
}

.emotion-4 {
  background-color: white;
  border-radius: .5rem;
  padding: 2rem 1rem;
  width: 80%;
  max-width: 360px;
}

.emotion-0 {
  margin: 0;
  font-weight: 500;
  text-align: center;
}

.emotion-2 {
  margin: 2.5rem 1rem;
}

.emotion-1 {
  border: none;
  border-bottom: 1px solid #ccc;
  padding: .5rem 1rem;
  font-size: 1.2rem;
  box-sizing: border-box;
  width: 100%;
  outline: none;
  -webkit-transition: border-bottom-color 300ms ease;
  transition: border-bottom-color 300ms ease;
}

.emotion-1:focus {
  border-bottom-color: #7E00F3;
}

.emotion-3 {
  background-color: #7E00F3;
  border-radius: .5rem;
  color: white;
  width: 100%;
  padding: .75rem .5rem;
  box-sizing: border-box;
  font-size: .9rem;
  font-weight: 500;
  text-transform: uppercase;
  outline: none;
}

<main
    class="emotion-5"
  >
    <form
      class="emotion-4"
    >
      <h1
        class="emotion-0"
      >
        Welcome to Tutopia
      </h1>
      <div
        class="emotion-2"
      >
        <input
          aria-label="Enter your username"
          class="emotion-1"
          id="username"
          name="username"
          placeholder="sergio"
          type="text"
        />
      </div>
      <button
        class="emotion-3"
        type="submit"
      >
        Login In
      </button>
    </form>
  </main>
</DocumentFragment>
`;
